src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -110,12 +110,12 @@ public override Task<bool> HandleRequestAsync()
             var remoteSignOutContext = new RemoteSignOutContext(Context, Scheme, Options, message);
             await Events.RemoteSignOut(remoteSignOutContext);
 
-            if (remoteSignOutContext.HandledResponse)
+            if (remoteSignOutContext.State == EventResultState.HandleResponse)
             {
                 Logger.RemoteSignOutHandledResponse();
                 return true;
             }
-            if (remoteSignOutContext.Skipped)
+            if (remoteSignOutContext.State == EventResultState.SkipToNextMiddleware)
```

Comment:
> which is why it does require a lot of code

Except this scenario is quite frequent and yet is still extremely painful: `ExternalLoginSignInAsync` doesn't take a `ClaimsPrincipal`/claims collection so you pretty much have to rewrite everything it internally calls: `SignInOrTwoFactorAsync`, `StoreTwoFactorInfo` and `SignInAsync`. It's a total nightmare for such a simple thing.